homeassistant/components/tplink_omada/sensor.py

```diff
@@ -0,0 +1,148 @@
+    DeviceStatus.MANAGED_EXTERNALLY: OmadaDeviceStatus.MANAGED_EXTERNALLY,
+    DeviceStatus.MANAGED_EXTERNALLY_WIRELESS: OmadaDeviceStatus.MANAGED_EXTERNALLY,
+}
+
+# High level status categories, suitable for most device statuses.
+DEVICE_STATUS_CATEGORY_MAP = {
+    DeviceStatusCategory.DISCONNECTED: OmadaDeviceStatus.DISCONNECTED,
+    DeviceStatusCategory.CONNECTED: OmadaDeviceStatus.CONNECTED,
+    DeviceStatusCategory.PENDING: OmadaDeviceStatus.PENDING,
+    DeviceStatusCategory.HEARTBEAT_MISSED: OmadaDeviceStatus.HEARTBEAT_MISSED,
+    DeviceStatusCategory.ISOLATED: OmadaDeviceStatus.ISOLATED,
+}
+
+
+def _map_device_status(device: OmadaListDevice) -> str:
+    """Map the API device status to the best available descriptive device status."""
+    return DEVICE_STATUS_MAP.get(
+        device.status,
+        DEVICE_STATUS_CATEGORY_MAP.get(
+            device.status_category, OmadaDeviceStatus.UNKNOWN
+        ).value,
+    )
+
+
+async def async_setup_entry(
+    hass: HomeAssistant,
+    config_entry: OmadaConfigEntry,
+    async_add_entities: AddEntitiesCallback,
+) -> None:
+    """Set up sensors."""
+    controller = config_entry.runtime_data
+
+    devices_coordinator = controller.devices_coordinator
+
+    entities: list[OmadaDeviceEntity] = []
+    for device in devices_coordinator.data.values():
+        entities.extend(
+            OmadaDeviceSensor(devices_coordinator, device, desc)
+            for desc in OMADA_DEVICE_SENSORS
+            if desc.exists_func(device)
+        )
+
+    async_add_entities(entities)
+
+
+@dataclass(frozen=True, kw_only=True)
+class OmadaDeviceSensorEntityDescription(SensorEntityDescription):
+    """Entity description for a status derived from an Omada device in the device list."""
+
+    exists_func: Callable[[OmadaListDevice], bool] = lambda _: True
+    update_func: Callable[[OmadaListDevice], StateType]
+
+
+OMADA_DEVICE_SENSORS: list[OmadaDeviceSensorEntityDescription] = [
+    OmadaDeviceSensorEntityDescription(
+        key="device_status",
+        translation_key="device_status",
+        device_class=SensorDeviceClass.ENUM,
+        entity_category=EntityCategory.DIAGNOSTIC,
+        update_func=_map_device_status,
+        options=[v.value for v in OmadaDeviceStatus],
+        has_entity_name=True,
+    ),
+    OmadaDeviceSensorEntityDescription(
+        key="cpu_usage",
+        translation_key="cpu_usage",
+        entity_category=EntityCategory.DIAGNOSTIC,
+        state_class=SensorStateClass.MEASUREMENT,
+        native_unit_of_measurement="%",
+        update_func=lambda device: device.cpu_usage,
+        has_entity_name=True,
+    ),
+    OmadaDeviceSensorEntityDescription(
+        key="mem_usage",
+        translation_key="mem_usage",
+        entity_category=EntityCategory.DIAGNOSTIC,
+        state_class=SensorStateClass.MEASUREMENT,
+        native_unit_of_measurement="%",
+        update_func=lambda device: device.mem_usage,
+        has_entity_name=True,
```

Comment:
Don't all your entities already have has entity name true? Might as well just set it on the top most class

Reply:
Do you mean in the entity description class definition or in the `OmadaDeviceEntity[_T]` base class for entities?

Reply:
Move it to the class with `_attr_has_entity_name = True`